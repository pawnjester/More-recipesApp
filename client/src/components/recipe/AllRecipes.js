import React, { Component } from 'react';
import Dropzone from 'react-dropzone';
import sha1 from 'sha1';
import superagent from 'superagent';

class AllRecipes extends Component {

  constructor(){
    super()
    this.state = {
      images: []
    }
  }

  upLoadFile(files) {
    console.log('uploadFile: ')
    const image = files[0]

    const cloudName = 'digr7ls7o'

    const url =  'https://api.cloudinary.com/v1_1/'+cloudName+'/image/upload'

    const timestamp = Date.now()/1000
    const uploadPreset = 'bd7kolap'

    const paramStr ='timestamp='+timestamp+'&upload_preset='+uploadPreset+'mr9XYhDjoiITDCHXlNm8jEvV03w';

    const signature = sha1(paramStr)

    const params = {
      'api_key': '569938115268323',
      'timestamp' : timestamp,
      'upload_preset': uploadPreset,
      'signature': signature
    }

    let uploadRequest = superagent.post(url)
    uploadRequest.attach('file', image)

    Object.keys(params).forEach((key) => {
      uploadRequest.field(key, params[key])
    })

    uploadRequest.end((err, resp) => {
      if(err) {
        // alert(err)
        return
      }

      console.log('UPLOAD COMPLETE: '+JSON.stringify(resp.body))
      const uploaded = resp.body

      let updatedImages = Object.assign([], this.state.images)
      updatedImages.push(uploaded)

      this.setState({
        images: updatedImages
      })
    })
  }
  render() {
    const list = this.state.images.map((image, i) => {
      return (
        <li key={i}>
          <img src= {image.secure_url} style = {{width: 50, height: 50}}/>
        </li>
      )
    })

    return (
      <div style ={{paddingTop : 100}}>
        Images Component <br/>
        <button name="myFile" type="file" onClick={this.upLoadFile.bind(this)} />
        <ol>
          { list }
        </ol>
        
      </div>
      
    )
  }





}

export default AllRecipes;