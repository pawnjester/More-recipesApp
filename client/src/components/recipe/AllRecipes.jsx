import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import ReactPaginate from 'react-paginate';
import NavigationBar from '../NavigationBar';
import Loader from '../common/Loader';
import getAllRecipes from '../../actions/getAllRecipes';
import Footer from '../common/Footer';
import Recipe from './RecipeCard';
/**
 * @description All Recipes component
 *
 * @class AllRecipes
 *
 * @extends {Component}
 */
class AllRecipes extends Component {
  /**
   * @description Creates an instance of AllRecipes.
   *
   * @constructor
   *
   * @param {any} props
   *
   * @memberof AllRecipes
   *
   * @returns {void}
   */
  constructor(props) {
    super(props);
    this.state = {
      loading: true,
    };
    this.onPageChange = this.onPageChange.bind(this);
  }
  /**
 * @description get All recipes
 *
 *@method
 *
 * @memberof AllRecipes
 *
 * @returns {void}
 */
  componentWillMount() {
    this.props.getAllRecipes()
      .then(() => {
        this.setState({ loading: false });
      });
  }

  /**
 * @description set state to currentPage
 *
 * @param {any} data
 *
 * @memberof FavoriteRecipe
 *
 * @returns {void}
 */
  onPageChange(data) {
    data.selected += 1;
    this.props.getAllRecipes(data.selected);
  }

  /**
 *
 * @description renders the jsx element
 *
 * @memberof RecipePage
 *
 * @returns {void}
 */
  render() {
    const recipeList = this.props.allRecipes.recipes || [];
    const pageCount = this.props.allRecipes.Pages;
    return (
      <div>
        <NavigationBar search="true" />
        <div className="container text-center">
          <div className="heading">
            <h1 className="p-5 ">All Recipes</h1>
          </div>
          <hr />
          <div className="row high">
            {this.state.loading ?
              <Loader Loading={this.state.loading} /> :
              recipeList.length > 0 ?
              recipeList.map(recipe =>
                (<Recipe
                  recipe={recipe}
                  key={recipe.id}
                />)) : (<h4 className="mt-5 text-center no-recipes"> No recipes yet </h4>)}
          </div>
        </div>
        {recipeList.length > 0 && <ReactPaginate
          pageCount={pageCount}
          pageRangeDisplayed={5}
          marginPagesDisplayed={3}
          previousLabel="Previous"
          nextLabel="Next"
          breakClassName="text-center"
          initialPage={0}
          containerClassName="container pagination justify-content-center"
          pageClassName="page-item"
          pageLinkClassName="page-link"
          activeClassName="page-item active"
          previousClassName="page-item"
          nextClassName="page-item"
          nextLinkClassName="page-link"
          previousLinkClassName="page-link"
          onPageChange={this.onPageChange}
        />}

        <Footer />
      </div>
    );
  }
}

AllRecipes.propTypes = {
  getAllRecipes: PropTypes.func.isRequired
};
const mapStateToProps = state => ({
  allRecipes: state.recipeReducer.recipes
});

export default connect(mapStateToProps, { getAllRecipes })(AllRecipes);
