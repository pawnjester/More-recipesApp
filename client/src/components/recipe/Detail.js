import React, { Component } from 'react'

export default class Detail extends Component {
  render() {
    return (
        <div class ="container home-inner">
          <div class= "row justify-content-center">            
            <div class='col-lg-6 '>
              <div class= "card bg-danger text-center card-form">
                <div class="card-body">
                  <h3>Sign In</h3>
                  <p>Welcome to More-Recipes, Please log in</p>
                  <form>
                    
                    <div class= "form-group">
                      <input type="text" class="form-control form-control-lg" placeholder="Username"/>
                      
                    </div>
                    <div class= "form-group">
                      <input type="password" class="form-control form-control-lg" placeholder="Password"/>
                      
                    </div>
                    
                    <input type="submit" value ="Submit" class="btn btn-outline-light btn-block"/>
                  </form>
                  <div class="card-body ">                    
                    <p class="card-text text-right"><a href="reset_password.html"  class="text-white">Forgot Password?</a> </p>                    
                  </div>
                </div>
            </div>
            
          </div>
          
        </div>
        
      </div>
      
    )
}
}
