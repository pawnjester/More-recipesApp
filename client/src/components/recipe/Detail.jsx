import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import toastr from 'toastr';
import NavigationBar from '../NavigationBar';
import '../../styles/detail.scss';
import getRecipeDetail from '../../actions/getRecipeDetail';
import getReview from '../../actions/getAllReviews';
import upvoteRecipe from '../../actions/upvoteRecipe';
import downvoteRecipe from '../../actions/downvoteRecipe';
import favoriteRecipe from '../../actions/favoriteRecipe';
import getUserDetail from '../../actions/getUserDetail';
import deleteReview from '../../actions/deleteReview';
import getFavoriteId from '../../actions/checkFavoriteId';
import Review from './Review';
import DisplayReviews from './DisplayReviews';
import Footer from '../common/Footer';
/**
 *
 * @class Detail
 *
 * @extends {Component}
 */
class Detail extends Component {
  /**
   *@description Creates an instance of Detail.
   *
   * @param {any} props
   *
   * @memberof Detail
   */
  constructor(props) {
    super(props);

    this.state = {
      message: '',
    };
    this.upvoted = this.upvoted.bind(this);
    this.onDeleteReview = this.onDeleteReview.bind(this);
  }
  /**
 *@description get recipe detail and user detail
 *
 * @memberof Detail
 *
 * @returns {void}
 */
  componentWillMount() {
    this.props.getRecipeDetail(this.props.match.params.recipeId);
    this.props.getUserDetail();
    this.props.getFavoriteId();
  }
  /**
 *@description set props to state
 *
 * @param {any} nextProps
 *
 * @memberof Detail
 *
 * @returns {void}
 */
  componentWillReceiveProps(nextProps) {
    this.setState({
      message: nextProps.message
    });
  }

  /**
 *
 * @param {any} reviewId
 *
 * @memberof Detail
 *
 * @returns {void}
 */
  onDeleteReview(reviewId) {
    this.props.deleteReview(reviewId);
  }

  /**
 *@description upvoted action
 *
 * @memberof Detail
 *
 * @returns {void}
 */
  upvoted() {
    this.props.upvoteRecipe(this.props.match.params.recipeId);
  }
  /**
 *@description favorite recipe
 *
 * @memberof Detail
 *
 * @returns {void}
 */
  favorite() {
    this.props.favoriteRecipe(this.props.match.params.recipeId)
      .then(() => {
        toastr.success(`${this.props.message}`);
      });
  }
  /**
 *@description downvote a recipe
 *
 * @memberof Detail
 *
 * @returns {void}
 */
  downvoted() {
    this.props.downvoteRecipe(this.props.match.params.recipeId);
  }
  /**
 *@description renders jsx element
 *
 * @memberof Detail
 *
 * @returns {void}
 */
  render() {
    const {
      singleRecipe, checkIfFavorite
    } = this.props;
    const ingredientss = singleRecipe.ingredients ? singleRecipe.ingredients : '';
    const methods = singleRecipe.method ? singleRecipe.method : '';
    const splittedMethod = methods.split(',');
    const splittedIngredients = ingredientss.split(',');
    const checkFavorited = checkIfFavorite || [];
    const { message } = this.state;
    const user = this.props.userDetail || {};
    const single = this.props.singleRecipe.Reviews || [];
    const style = {
      backgroundImage: `url(${singleRecipe.imageUrl})`,
    };

    return (
      <div>
        <NavigationBar />
        <div className="header-banner" style={style} />

        <div className="container detail-container bg-white">
          <div className="row">
            <div className="col-md-8 col-sm-12" >
              <img className="card-img-top img-fluid" src={singleRecipe.imageUrl} alt="Card cap" />
            </div>
            <div className="col-md-4 col-sm-12">
              <div className="detail-holder">
                <h1 className="detail-title">{singleRecipe.name}</h1>
                <i className="fa fa-eye" aria-hidden="true" /><span id="clickableAwesomeFont" className="view" >&nbsp;{singleRecipe.viewCount}</span>
                <div className="card detail-card">
                  <div className="card-body clearfix">
                    <div className="row">
                      <button className="btn btn-success col-sm-4 no-border-r pt-3 pb-3" onClick={() => this.upvoted()}>
                        <i className="fa fa-thumbs-up" aria-hidden="true" /><span>&nbsp;{singleRecipe.upVotes}</span>
                      </button>
                      <button className="btn btn-danger col-sm-4 no-border-r pt-3 pb-3" onClick={() => this.downvoted()}>
                        <i className="fa fa-thumbs-down" aria-hidden="true" /><span>&nbsp;{singleRecipe.downVotes}</span>
                      </button>
                      <button className="btn btn-white col-sm-4 no-border-x pt-3 pb-3" onClick={() => this.favorite()}>
                        <i
                          className="fa fa-heart"
                          aria-hidden="true"
                          style={checkFavorited.includes(singleRecipe.id) ? { color: 'red' } : { color: 'grey' }}
                        /><span>&nbsp;</span>
                      </button>
                    </div>
                  </div>
                  <div className="card-body">
                    <h6>Cooking Time</h6>
                    <p><i className="fa fa-clock-o mr-2" aria-hidden="true" style={{ color: 'orange', fontSize: 25 }} />{singleRecipe.cookingTime}</p>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div className="container detail-height">
            <div className="row">
              <div className="col-md-6 col-sm-12 mt-5">
                <h4 className="text-center">Preparation</h4>
                <ul className="list-group wrap-word">
                  {
                  splittedMethod.map(split =>
                    <li className="list-group-item">{split}</li>)
                }
                </ul>
              </div>
              <div className="col-md-6 col-sm-12 mt-5">
                <h4 className="text-center">Ingredients</h4>
                <ul className="list-group wrap-word">
                  { splittedIngredients.map(split =>
                    <li className="list-group-item" key={split.id}>{ split }</li>)
                }
                </ul>
              </div>

            </div>
          </div>
          <Review recipeId={singleRecipe.id} />
          <div className="container mb-5 mt-5">
            <h1>Reviews</h1>
            <hr />
            {single.length < 1 && (<h4 className="mt-5 text-center" > No reviews </h4>)}
            {single.map(reviewed =>
              <DisplayReviews key={reviewed.id} Review={reviewed} deleteReview={this.onDeleteReview} userDetail={user} />)}
          </div>
        </div>
        <Footer />
      </div>


    );
  }
}

Detail.propTypes = {
  getRecipeDetail: PropTypes.func.isRequired,
  upvoteRecipe: PropTypes.func.isRequired,
  downvoteRecipe: PropTypes.func.isRequired,
  favoriteRecipe: PropTypes.func.isRequired,
  deleteReview: PropTypes.func.isRequired,
  getUserDetail: PropTypes.func.isRequired,
  getFavoriteId: PropTypes.func.isRequired
};
const mapStateToProps = state => ({
  singleRecipe: state.recipeDetailReducer.currentRecipe,
  favoriteRecipes: state.recipeDetailReducer.favoriteRecipe,
  message: state.recipeDetailReducer.message.message,
  userDetail: state.userDetailReducer.userDetail,
  checkIfFavorite: state.recipeDetailReducer.checkIfFavorited
});

export default connect(mapStateToProps, {
  getRecipeDetail, getReview, upvoteRecipe, downvoteRecipe, favoriteRecipe, getUserDetail, deleteReview, getFavoriteId
})(Detail);
