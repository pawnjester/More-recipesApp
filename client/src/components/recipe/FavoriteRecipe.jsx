import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import JwtDecode from 'jwt-decode';
import NavigationBar from '../NavigationBar';
import '../../styles/favorite.scss';
import GetFavoriteRecipe from '../../actions/getFavoriteRecipes';
import DeleteFavorite from '../../actions/deleteFavorite';
import SingleFavorite from './SingleFavorite';
import Footer from '../common/Footer';
/**
 *
 *
 * @class FavoriteRecipe
 * @extends {Component}
 */
class FavoriteRecipe extends Component {
  constructor(props) {
    super(props);

    this.onDelete = this.onDelete.bind(this);
  }
/**
 *
 *
 * @memberof FavoriteRecipe
 */
  componentWillMount() {
    const userId = JwtDecode(localStorage.jwtToken);
    this.props.GetFavoriteRecipe(userId.id);
  }

  onDelete(favoriteId) {
    this.props.DeleteFavorite(favoriteId);
  }
  /**
 *
 *
 * @returns
 * @memberof FavoriteRecipe
 */
  render() {
    const favoriteRecipe = this.props.favoriteRecipe ? this.props.favoriteRecipe : [];

    return (
      <div>
        <NavigationBar />
        <div className="container text-center  ">
          <div className="heading">
            <h1 className="p-5 ">Favorite Recipes</h1>
          </div>
          <hr />
          <div className="row high">
            {favoriteRecipe.map(favorite =>
              <SingleFavorite recipe={favorite.Recipe} deleteFavorite={this.onDelete} favoriteId={favorite.id} key={favorite.Recipe.id} />)}
          </div>
        </div>
        <Footer />
      </div>
    );
  }
}

FavoriteRecipe.propTypes = {
  GetFavoriteRecipe: PropTypes.func.isRequired,
  favoriteRecipe: PropTypes.arrayOf(PropTypes.object).isRequired,
};

const mapStateToProps = state => ({
  favoriteRecipe: state.recipeDetailReducer.favoriteRecipes,
});

export default connect(
  mapStateToProps,
  {
    GetFavoriteRecipe,
    DeleteFavorite
  },
)(FavoriteRecipe);
