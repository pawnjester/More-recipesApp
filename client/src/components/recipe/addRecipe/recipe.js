import React, { Component } from 'react';

export default class Recipe extends Component {
  constructor(props) {
    super(props)
    this.onDelete = this.onDelete.bind(this);
  }

  onDelete =(event) => {
    console.log('we are deleting');                          
    this.props.deleteRecipe(this.props.recipe.id);
    console.log('we are deleting******');                
  }
  render() {
    return (
      <div className="col-md-4 col-xs-12 " >
      
        <div className="card">
            <img className="card-img-top" src= {this.props.recipe.imageUrl}/>
            <div className="card-body">
              <h4 className="card-title">{this.props.recipe.name}</h4>
              <p className="card-text">{this.props.recipe.ingredients}</p>
              <p className="card-text">{this.props.recipe.method}</p>

            </div>
            <div className="card-body clearfix">
              <div className="text-left text-success float-left">
                <i className="fa fa-pencil" aria-hidden="true"></i><a href="" data-toggle="modal" data-target="#editModal"><span id='clickableAwesomeFont' className="text-success">&nbsp;Edit</span></a>
              </div>
              <div className="text-right text-danger float-right">
                <i className="fa fa-trash" aria-hidden="true"></i><span id='clickableAwesomeFont' onClick = {() =>this.onDelete()}className="delete">&nbsp;Delete</span>
              </div>
              <div className="text-center text-primary float-center">
                <i className="fa fa-eye" aria-hidden="true"></i><span id='clickableAwesomeFont' className="view">&nbsp;View</span>
              </div>
            </div>
          </div>
          </div>
          
    )
  }
}
