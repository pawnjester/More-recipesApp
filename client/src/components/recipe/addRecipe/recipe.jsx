import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import EditRecipeModal from '../../Modal/EditRecipeModal';


class Recipe extends Component {
  constructor(props) {
    super(props);
    this.onDelete = this.onDelete.bind(this);

    this.state = {
      modal: false,
    };
    this.toggle = this.toggle.bind(this);
  }
  onDelete() {
    this.props.deleteRecipe(this.props.recipe.id);
  }

  toggle() {
    this.setState({
      modal: !this.state.modal,
    });
  }

  render() {
    const style = {
      height: 200,
    };
    const { recipe } = this.props;
    return (
      <div className="col-md-4 col-xs-12 " >
        <div className="card">
          <img className="card-img-top" src={recipe.imageUrl} style={style} alt="recipeImage" />
          <div className="card-body">
            <h4 className="card-title">{recipe.name}</h4>
            <p className="card-text">{recipe.ingredients}</p>
            <p className="card-text">{recipe.method}</p>

          </div>
          <div className="card-body clearfix">
            <div className="text-left text-success float-left">
              <i className="fa fa-pencil" aria-hidden="true" />
              <span id="clickableAwesomeFont" onClick={this.toggle} >&nbsp;Edit</span>
              <EditRecipeModal
                recipe={recipe}
                isOpen={this.state.modal}
                toggle={this.toggle}
                // editRecipe={this.props.editRecipe}
                id={this.props.recipe.id}
                getAllRecipes={this.props.getAllRecipes}
              />
            </div>

            <div className="text-right text-danger float-right">
              <i className="fa fa-trash" aria-hidden="true" /><span id="clickableAwesomeFont" onClick={() => this.onDelete()} className="delete">&nbsp;Delete</span>
            </div>
            <Link to={`/detail/${recipe.id}`}>
              <div className="text-center text-primary float-center">
                <i className="fa fa-eye" aria-hidden="true" /><span id="clickableAwesomeFont" className="view" >&nbsp;View</span>
              </div>
            </Link>

          </div>
        </div>
      </div>
    );
  }
}

export default Recipe;
