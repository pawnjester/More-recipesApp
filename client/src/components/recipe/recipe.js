import React, { Component } from 'react'

class Recipe extends Component {
  render() {
    return (
        <div class="row high">
      <div class="col-md-4 col-xs-4">
        <div class="card">
          <img class="card-img-top" src="img/chcolate-cream.jpg" alt="Chocolate cream"/>
          <div class="card-body">
            <h4 class="card-title">Chocolate cream</h4>
            <p class="card-text">Some quick example text to build on the card title and make up the bulk of the card's content.</p>

          </div>
          <div class="card-body clearfix">
            <div class="text-left text-success float-left">
              <i class="fa fa-pencil" aria-hidden="true"></i><a href="" data-toggle="modal" data-target="#editModal"><span id='clickableAwesomeFont' class="text-success">&nbsp;Edit</span></a>
            </div>
            <div class="text-right text-danger float-right">
              <i class="fa fa-trash" aria-hidden="true"></i><span id='clickableAwesomeFont' class="delete">&nbsp;Delete</span>
            </div>
            <div class="text-center text-primary float-center">
              <i class="fa fa-eye" aria-hidden="true"></i><span id='clickableAwesomeFont' class="view" href="detail.html">&nbsp;View</span>
            </div>
          </div>
        </div>
      </div>
      </div>
    )
  }
}

export default Recipe;
