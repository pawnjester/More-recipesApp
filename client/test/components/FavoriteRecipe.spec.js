import React from 'React';
import { shallow, mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import expect from 'expect';
import sinon from 'sinon';
import { provider } from 'react-redux';
import { FavoriteRecipe } from '../../src/components/recipe/FavoriteRecipe';
import store from '../../src/store/store';

configure({ adapter: new Adapter() });

const props = {
  getFavoriteRecipe: jest.fn(),
  deleteFavorite: jest.fn(),
  favoriteRecipe: {
    id: 1,
    name: 'Rice',
    method: 'Boil the rice',
    upVotes: 1,
    downVotes: 0
  }
};

const setup = () => {
  const shallowWrapper = shallow(<FavoriteRecipe {...props}/> );
  return {
    shallowWrapper,
  }
};

describe('Favorite Recipe Component', () => {
  it('should render correctly', () => {
    const { shallowWrapper } = setup();
    expect(shallowWrapper).toMatchSnapshot();
  });
});

describe('onDelete() should', () => {
  it('be called when form is submitted', () => {
    const { shallowWrapper } = setup();
    const favoriteId = 1;
    shallowWrapper.instance().onDelete(favoriteId);
    expect(props.deleteFavorite.mock.calls.length).toEqual(1);
});
});

describe('onPageChange', () => {
  it('should work correctly', () => {
    sinon.spy(FavoriteRecipe.prototype, 'onPageChange');
    const props = {
      getFavoriteRecipe: jest.fn(),
      deleteFavorite: jest.fn(),
      favoriteRecipe: {
        id: 1,
        name: 'Rice',
        method: 'Boil the rice',
        upVotes: 1,
        downVotes: 0
      }
    };
    const selected = {
      data: 1
    };
    const shallowWrapper = shallow(<FavoriteRecipe {...props} store={store} />);
    shallowWrapper.instance().onPageChange(selected);
    expect(shallowWrapper.instance().props.getFavoriteRecipe).toBeCalled();
  });
});
