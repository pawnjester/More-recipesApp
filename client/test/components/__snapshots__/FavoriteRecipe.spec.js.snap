// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorite Recipe Component should render correctly 1`] = `
<div>
  <Connect(NavigationBar)
    search="true"
  />
  <div
    className="container text-center  "
  >
    <div
      className="heading"
    >
      <h1
        className="p-5 "
      >
        Favorite Recipes
      </h1>
    </div>
    <hr />
    <div
      className="row high"
    >
      <h4
        className="mt-5 text-center no-favorite"
      >
        You have no favorite
      </h4>
    </div>
  </div>
  <PaginationBoxView
    activeClassName="page-item active"
    breakClassName="text-center"
    breakLabel="..."
    containerClassName="container pagination justify-content-center"
    disableInitialCallback={false}
    disabledClassName="disabled"
    initialPage={0}
    marginPagesDisplayed={3}
    nextClassName="page-item"
    nextLabel="Next"
    nextLinkClassName="page-link"
    onPageChange={[Function]}
    pageClassName="page-item"
    pageCount={10}
    pageLinkClassName="page-link"
    pageRangeDisplayed={5}
    previousClassName="page-item"
    previousLabel="Previous"
    previousLinkClassName="page-link"
  />
  <Footer />
</div>
`;
